diff -up openwsman-2.2.3/etc/init/openwsmand.sh.in.orig openwsman-2.2.3/etc/init/openwsmand.sh.in
--- openwsman-2.2.3/etc/init/openwsmand.sh.in.orig	2010-01-21 11:12:34.000000000 +0100
+++ openwsman-2.2.3/etc/init/openwsmand.sh.in	2010-03-02 14:29:31.618984202 +0100
@@ -4,15 +4,13 @@
 # Provides: openwsmand
 # Required-Start: $remote_fs
 # Required-Stop: $network
-# Default-Start: 2 3 4 5
-# Default-Stop: 0 1 6 
 # Short-Description: Openwsman Daemon
 # Description: openwsmand
 #       Start/Stop the Openwsman Daemon
 ### END INIT INFO
 #
 #
-# chkconfig: 2345 36 64
+# chkconfig: - 36 64
 # description: Openwsman Daemon
 # processname: openwsmand
 
@@ -61,7 +59,7 @@ start()
 	if [ "x${FQDN}" = "x" ]; then
 	  FQDN=localhost.localdomain
 	fi
-cat << EOF | sh @SYSCONFDIR@/owsmangencert.sh > /dev/null 2>&1
+cat << EOF | sh @libexecdir@/openwsman/owsmangencert.sh > /dev/null 2>&1
 --
 SomeState
 SomeCity
